sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1064 @@
+ * </p>
+ * <p>
+ * Using these annotations a programmer can declare how the code is meant to behave, and static code
+ * analysis (like e.g. FindBugs) can be used to verify that this is actually true. Also these
+ * annotations help others understanding code more easily, e.g. if confrontend with an annotated
+ * interface the necessity for null checks can easily be deducted from the annotations.
+ * </p>
+ * <p>
+ * The Jsr305AnnotationsCheck supports enforcing the following code style:
+ * </p>
+ * <ul>
+ * <li>Every method declaration, implementation or lambda requires nullness annotations for all
+ * parameters and return values except for primitive types (because a void or an int can never be
+ * null anyway).</li>
+ * <li>The annotation can be made directly or applied through either inheritance from an already
+ * annotated class or a annotation for class-wide defaults.</li>
+ * <li>Annotations need to make sense. For instance, a class-scope annotation cannot be used for a
+ * method, and a method annotation cannot be used for a class.</li>
+ * <li>In overridden methods, the following rule applies (regardless of what was annotated in the
+ * parent method): For parameter definitions {@code @Nonnull} annotation is always illegal because
+ * being less "nullable" cannot be assumed for a parameter in an inherited method. Conversely
+ * {@code @Nullable} is always allowed. For return values it is the other way round:
+ * {@code @CheckForNull} is always illegal while {@code @Nonnull} is always legal.</li>
+ * </ul>
+ * <p>
+ * The following configuration properties are supported:
+ * </p>
+ * <dl>
+ * <dt>{@code packages = com.github.sevntu.pkg1,com.github.sevntu.pkg2}</dt>
+ * <dd>Activate this check for a list of parent packages and their children.</dd>
+ * <dt>{@code excludePackages = com.github.sevntu.pkg1.sub1,com.github.sevntu.pkg1.sub2}</dt>
+ * <dd>Set packages excluded from checking. This setting can be useful if under the parent package
+ * set with "packages" there are subpackages which should not be checked.</dd>
+ * <dt>{@code allowOverridingReturnValue = true}</dt>
+ * <dd>Annotating return values "@CheckForNull" in overridden methods is flagged as an error. When
+ * setting the this property to true, this error will be ignored (useful for upgrading).</dd>
+ * <dt>{@code allowOverridingParameters = true}</dt>
+ * <dd>Annotating parameters "@Nonnull" in overridden methods is flagged as an error. When setting
+ * this property to true, this error will be ignored (useful for upgrading).</dd>
+ * </dl>
+ * <p>
+ * Example code:
+ * </p>
+ *
+ * <pre>
+ * // no class-level annotation
+ * class Class1 {
+ *     &#64;CheckForNull // Error: obj2 not annotated!
+ *     String method1(@Nullable Object obj1, Object obj2) { ... }
+ *
+ *     // Error: return value not annotated
+ *     String method2() { ... }
+ * }
+ *
+ * // with class-level defaults
+ * &#64;ParametersAreNonnullByDefault
```

Comment:
please update examples to be like we do in main project - https://github.com/checkstyle/checkstyle/blob/3767e429133d223ca11c385ad3f802a47472f33c/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java#L126

format:
1. brief explanation in words
2. exact config
3. exact code (minimized, close to compile state) with trailing line comment where violation should be.

Reply:
I have added some comments with the next commit, please check whether this is OK for you.